lib/statsd/instrument/compiled_metric.rb

```diff
@@ -187,19 +220,18 @@ def build(name:, type:, value_format:, client_prefix:, no_prefix:, default_tags:
 
             # Build the datagram blueprint
             # Format: "<prefix><name>:<value_format>|<type>|@<sample_rate>|#<tags>"
-            # value_format is delegated to the metric subclass (e.g., %d for Counter, %f for others)
             # Note: When aggregation is enabled, sample_rate is applied before aggregation
             if sample_rate && sample_rate < 1
               # Include sample_rate in the blueprint (only when not aggregating)
               if all_tags.empty?
-                "#{prefix}#{normalized_name}:#{value_format}|#{type}|@#{sample_rate}"
+                "#{prefix}#{normalized_name}:%s|#{type}|@#{sample_rate}"
               else
-                "#{prefix}#{normalized_name}:#{value_format}|#{type}|@#{sample_rate}|##{all_tags}"
+                "#{prefix}#{normalized_name}:%s|#{type}|@#{sample_rate}|##{all_tags}"
               end
             elsif all_tags.empty?
-              "#{prefix}#{normalized_name}:#{value_format}|#{type}"
+              "#{prefix}#{normalized_name}:%s|#{type}"
             else
-              "#{prefix}#{normalized_name}:#{value_format}|#{type}|##{all_tags}"
+              "#{prefix}#{normalized_name}:%s|#{type}|##{all_tags}"
```

Comment:
I was checking in rbi and I think we should not use `%s` here. It might cause strange behavior when printing very small floats (think cache operations or smaller operations). Which will cause the values/samples to be dismissed.

See: 

```
irb(main):007> my_f_seconds = 0.00000201214124
=> 2.01214124e-06
irb(main):008> my_f_seconds.to_s
=> "2.01214124e-06"
irb(main):009> blueprint % my_f_seconds
=> "metric:2.01214124e-06|c|#foo:bar"
irb(main):010> my_f_seconds = 0.0000201214124
=> 2.01214124e-05
irb(main):011> my_f_seconds.class
=> Float
irb(main):012> arr_vals = [0.000000123,0.123123123,0.0000124123]
=> [1.23e-07, 0.123123123, 1.24123e-05]
irb(main):013* arr_vals.join(":")
=> "1.23e-07:0.123123123:1.24123e-05"
```